projects/vdk-core/src/vdk/internal/builtin_plugins/builtin_hook_impl.py

```diff
@@ -50,24 +50,36 @@ def vdk_initialize(self, context: CoreContext) -> None:
 
         Configuration can override an attempt/op/execution id.
 
-        Otherwise each is auto-generated in format:
-        * Attempt id format is "execution_id-random_suffix"
-        * Execution id format is "op_id-random_suffix"
-        * Op Id is random string.
+        Attempt id and Execution id are extracted from env variables.
+        If not present they are auto generated.
+
+        If auto generated:
+        * Attempt id format is uuid + timestamp + five random chars.
+        * Execution id is attempt id minus last 6 chars.
+        * Op id is equal to execution id.
         """
         op_id = context.configuration.get_value(vdk_config.OP_ID)
         execution_id = context.configuration.get_value(vdk_config.EXECUTION_ID)
         attempt_id = context.configuration.get_value(vdk_config.ATTEMPT_ID)
 
-        if not op_id:
-            op_id = str(int(time.time()))
+        if not attempt_id:
+            # UUID is 36 chars
+            attempt_id = (
+                f"{str(uuid.uuid4())}-{str(int(time.time()))}-{str(uuid.uuid4())[:5]}"
+            )
```

Comment:
I think if EXECUTION_ID is provdied we should generate attempt id out of execution id . 
And only if it is not provided generate it from empty. 

Otherwise, We cannot be sure attempt id is in format [execution-id]-{5. chars}  which is what we want.


```suggestion
        if not attempt_id:
            if not execution_id:
                attempt_id = f"{execution_id}-{str(uuid.uuid4())[:5]}"
            else:
                # UUID is 36 chars
                attempt_id = (
                    f"{str(uuid.uuid4())}-{str(int(time.time()))}-{str(uuid.uuid4())[:5]}"
                )
```

Reply:
Shouldn't it be the other way around?

`        if not attempt_id:
            if not execution_id:
                                # UUID is 36 chars
                attempt_id = (
                    f"{str(uuid.uuid4())}-{str(int(time.time()))}-{str(uuid.uuid4())[:5]}"
                )
            else:
                attempt_id = f"{execution_id}-{str(uuid.uuid4())[:5]}"`

Reply:
Write unit tests and it would show. 

But the idea is if execution id is NOT empty to use it, otherwise generate it with `{str(uuid.uuid4())}`